tests/test_io.py

```diff
@@ -231,3 +233,31 @@ def test_spss():
     path = Path(CWD, "data", "labelled-num.sav")
     check(assert_type(read_spss(path, convert_categoricals=True), DataFrame), DataFrame)
     check(assert_type(read_spss(str(path), usecols=["VAR00002"]), DataFrame), DataFrame)
+
+
+def test_json():
+    with ensure_clean() as path:
+        check(assert_type(DF.to_json(path), None), type(None))
+        check(assert_type(read_json(path), DataFrame), DataFrame)
+    json_str = DF.to_json()
+    check(assert_type(json_str, str), str)
+    bin_json = io.StringIO(json_str)
+    check(assert_type(read_json(bin_json), DataFrame), DataFrame)
+
+
+def test_json_series():
+    s = DF["a"]
+    with ensure_clean() as path:
+        check(assert_type(s.to_json(path), None), type(None))
+        check(assert_type(read_json(path, typ="series"), Series), Series)
+    check(assert_type(DF.to_json(), str), str)
+
+
+def test_json_chunk():
+    with ensure_clean() as path:
+        check(assert_type(DF.to_json(path), None), type(None))
+        json_reader = read_json(path, chunksize=1, lines=True)
+        check(assert_type(json_reader, JsonReader), JsonReader)
+        for sub_df in json_reader:
+            check(assert_type(sub_df, Union[DataFrame, Series]), DataFrame)
```

Comment:
So we can make this a little better.  If you make `JsonReader` a `Generic` that allows either `Series` or `DataFrame`, then you know that `JsonReader.__next__()` will return that type, which means that you can change the two overloads to return `JsonReader[Series]` when `typ=="series"` and `JsonReader[DataFrame]` when `typ=="frame"`

Looking in the current pandas source, they did this, but used strings rather than the types as the arguments to `Generic`

Reply:
Made it generic `JsonReader[DataFrame]` or `JsonReader[Series]`.